src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -255,15 +259,27 @@ public void OnRemoved()
                 InMemoryStorage.DropCache(Analyzer);
             }
 
-            private sealed class AnalyzerTypeData
+            public bool SetCompilationEndAnalyzer(Project project, Compilation compilation)
```

Comment:
Split into two methods `bool GetIsCompilationEndAnalyzer` and `void ComputeIsCompilationEndAnalyzer`